test/sciml_style.jl

```diff
@@ -1434,4 +1493,540 @@
+        # custom alignment in typed arrays and dict literals, unlike YASStyle.
+        # The issue is that whitespace is converted to placeholders early in the
+        # formatting pipeline, making it difficult to preserve the original alignment
+        # even when delegating to YASStyle.
+
+        # The specific case from issue #935
+        str = raw"""
+        setup = Pair{String, Any}["Start time" => first(integrator.sol.prob.tspan),
+                                  "Final time" => last(integrator.sol.prob.tspan),
+                                  "time integrator" => integrator.alg |> typeof |> nameof,
+                                  "adaptive" => integrator.opts.adaptive]
+        """
+        expected = raw"""
+        setup = Pair{String, Any}["Start time" => first(integrator.sol.prob.tspan),
+                                  "Final time" => last(integrator.sol.prob.tspan),
+                                  "time integrator" => integrator.alg |> typeof |> nameof,
+                                  "adaptive" => integrator.opts.adaptive]
+        """
+        @test format_text(str, SciMLStyle()) == expected
+
+        # Test simpler typed arrays
+        str2 = raw"""
+        values = Int[1,
+                     2,
+                     3]
+        """
+        expected2 = raw"""
+        values = Int[1,
+                     2,
+                     3]
+        """
+        @test format_text(str2, SciMLStyle()) == expected2
+
+        # Another alignment test
+        str3 = raw"""
+        x = Float64[1.0,
+                    2.0,
+                    3.0]
+        """
+        expected3 = raw"""
+        x = Float64[1.0,
+                    2.0,
+                    3.0]
+        """
+        @test format_text(str3, SciMLStyle()) == expected3
+    end
+
+    @testset "Issue #934 comment #3146379860 - Additional edge cases" begin
+        # Test 1: Variable array indent
+        str = raw"""
+        kernels = [
+            GaussianKernel,
+            SchoenbergCubicSplineKernel,
+            SchoenbergQuarticSplineKernel,
+            SchoenbergQuinticSplineKernel,
+        ]
+        """
+        # With YAS-style alignment, arrays align with opening bracket
+        expected = raw"""
+        kernels = [GaussianKernel,
+                   SchoenbergCubicSplineKernel,
+                   SchoenbergQuarticSplineKernel,
+                   SchoenbergQuinticSplineKernel]
+        """
+        # Default style now uses YAS-style alignment
+        @test format_text(str, SciMLStyle()) == expected
+        
+        # With yas_style_nesting=true and existing line breaks, preserves them with standard indentation (PR #807)
+        expected_yas = raw"""
+        kernels = [GaussianKernel,
+            SchoenbergCubicSplineKernel,
+            SchoenbergQuarticSplineKernel,
+            SchoenbergQuinticSplineKernel]
+        """
+        @test format_text(str, SciMLStyle(), yas_style_nesting = true) == expected_yas
```

Comment:
```suggestion
        # With yas_style_nesting=true both versions should be valid and not be reformatted
        @test format_text(str, SciMLStyle(), yas_style_nesting = true) == str
        @test format_text(expected, SciMLStyle(), yas_style_nesting = true) == expected
```

Reply:
This is still open. The `expected_yas` version fits neither the YAS style (no line break + align) nor the variable indent style (line break + indent 4 spaces).